src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -3392,6 +3392,24 @@ static QVariant fcnIsClosed( const QVariantList &values, const QgsExpressionCont
   return QVariant::fromValue( curve->isClosed() );
 }
 
+static QVariant fcnCloseLine( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QgsGeometry geom = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
+
+  if ( geom.isNull() || geom.isMultipart() )
+    return QVariant();
+
+  const QgsLineString *line = qgsgeometry_cast< const QgsLineString * >( geom.constGet() );
+
+  if ( !line )
+    return QVariant();
+
+  QgsLineString *closed_line = line->clone();
```

Comment:
```suggestion
  std::unique_ptr< QgsLineString > closedLine( line->clone() );
```

Reply:
I was wondering how to do this correctly. My solution worked, but I was thinking that there is probably a more correct solution. Is there a specific reason why to handle it like this?